gateway-discovery-cm/src/main/java/org/apache/knox/gateway/topology/discovery/cm/ClouderaManagerServiceDiscovery.java

```diff
@@ -202,6 +204,22 @@ private static Cluster discoverCluster(DiscoveryApiClient client, String cluster
           }
         }
       }
+      /* Since Cloudera Manager does not have a service for itself, we will add a skeleton CM
+      service so that we can add CM service to topology when auto-discovery is
+      turned on and CM service is selected in the descriptor */
+      List<ServiceModelGenerator> smgList = serviceModelGenerators
+          .get(CM_SERVICE_TYPE);
+      if (smgList != null) {
+        for (ServiceModelGenerator serviceModelGenerator : smgList) {
+          if (serviceModelGenerator != null) {
+            serviceModelGenerator.setApiClient(client);
+            ServiceModel serviceModel = serviceModelGenerator
+                .generateService(null, null, null, null);
+            serviceModels.add(serviceModel);
+          }
+        }
+      }
+
```

Comment:
I believe it would be easier to read and understand later on if you add the CM service above to `serviceList` and modify `getRoles(rolesResourceApi, clusterName, serviceName)` to fetch the roles using the CM REST API for other services and add CM API/UI role manually. Then you would not have to copy-paste the logic which checks service model generators

Reply:
+1

Reply:
Sure, I'll update the PR with this change.